src/core/main.c

```diff
@@ -196,7 +193,27 @@ int main(int argc, char *argv[]) {
 	state.timer_poll_fd = pollfds[EVENT_LOOP_TIMER].fd;
 
 	while (1) {
-		ret = poll(pollfds, sizeof(pollfds) / sizeof(pollfds[0]), -1);
+		// sd-bus requires that we update FD/events/timeout every time we poll
+		pollfds[EVENT_LOOP_DBUS].fd = sd_bus_get_fd(state.bus);
+		if (pollfds[EVENT_LOOP_DBUS].fd < 0) {
+			logprint(ERROR, "sd_bus_get_fd failed: %s", strerror(errno));
```

Comment:
Nit: it doesn't seem like sd-bus populates errno. It seems like the error number is returned instead.